lib/std/array_list.zig

```diff
@@ -463,6 +463,7 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
         /// Asserts the list has at least one item.
         /// Invalidates pointers to the removed element.
         pub fn pop(self: *Self) T {
+            assert(self.items.len > 0);
```

Comment:
This assertion is redundant as the indexing on the next line is bounds checked in safe build modes and the bounds check checks the exact same condition.

The other assertions are redundant for the same reason.